lib/iris/io/loading.py

```diff
@@ -0,0 +1,603 @@
+            options = {}
+        elif isinstance(options, str) and options in self.SETTINGS:
+            options = self.SETTINGS[options]
+        elif not isinstance(options, Mapping):
+            msg = (
+                f"Invalid arg options={options!r} : "
+                f"must be a dict, or one of {tuple(self.SETTINGS.keys())}"
+            )
+            raise TypeError(msg)
+
+        # Override any options with keywords
+        options.update(**kwargs)
+        bad_keys = [key for key in options if key not in self.OPTION_KEYS]
+        if bad_keys:
+            msg = f"Unknown options {bad_keys} : valid options are {self.OPTION_KEYS}."
+            raise ValueError(msg)
+
+        # Implement all options by changing own content.
+        for key, value in options.items():
+            setattr(self, key, value)
+
+    def settings(self):
+        """Return an options dict containing the current settings."""
+        return {key: getattr(self, key) for key in self.OPTION_KEYS}
+
+    def __repr__(self):
+        msg = f"{self.__class__.__name__}("
+        msg += ", ".join(f"{key}={getattr(self, key)!r}" for key in self.OPTION_KEYS)
+        msg += ")"
+        return msg
+
+    @contextlib.contextmanager
+    def context(self, settings=None, **kwargs):
+        """Return a context manager applying given options.
+
+        Parameters
+        ----------
+        settings : str or dict
+            Options dictionary or name, as for :meth:`~LoadPolicy.set`.
+        kwargs : dict
+            Option values, as for :meth:`~LoadPolicy.set`.
+
+        Examples
+        --------
+        .. testsetup::
+
+            import iris
+            from iris import LOAD_POLICY, sample_data_path
+
+        >>> path = sample_data_path("time_varying_hybrid_height", "*.pp")
+        >>> with LOAD_POLICY.context("legacy"):
+        ...     cubes = iris.load(path, "x_wind")
+        >>> print(cubes)
+        0: x_wind / (m s-1)                    (time: 2; model_level_number: 5; latitude: 144; longitude: 192)
+        1: x_wind / (m s-1)                    (time: 12; model_level_number: 5; latitude: 144; longitude: 192)
+        2: x_wind / (m s-1)                    (model_level_number: 5; latitude: 144; longitude: 192)
+        >>>
+        >>> with LOAD_POLICY.context("recommended"):
+        ...     cubes = iris.load(path, "x_wind")
+        >>> print(cubes)
+        0: x_wind / (m s-1)                    (model_level_number: 5; time: 15; latitude: 144; longitude: 192)
+        """
+        # Save the current state
+        saved_settings = self.settings()
+
+        # Apply the new options and execute the context
+        try:
+            self.set(settings, **kwargs)
+            yield
+        finally:
+            # Re-establish the former state
+            self.set(saved_settings)
+
+
+#: A control object containing the current file loading options.
+LOAD_POLICY = LoadPolicy()
+
+
+def _combine_cubes(cubes, options, merge_require_unique):
```

Comment:
I guess for separation of concerns we could do that.  So we could separate out the 'combine' operation from `iris.io.loading` to create an `iris.combine`, if that seems preferable.
But as it's private for now, that doesn't really affect this PR.

Note that the _merge and _concatenate modules are private, so the public access to these is _only_ through the `CubeList` methods.
The[ follow-on PR](https://github.com/pp-mo/iris/pull/82/files) currently proposes to add `CubeList.combine[_cubes]` methods, but it does also publish the main function as `iris.io.loading.combine_cubes`.  
We can change that, but I'd prefer to address it separately.
I don't think this would affect the new `iris.io.loading` module, : that would anyway still have a place as it contains the main load_xxx functions and supporting routines.